utils/src/main/scala/za/co/absa/enceladus/utils/transformations/DeepArrayTransformations.scala

```diff
@@ -652,11 +789,84 @@ object DeepArrayTransformations {
 
   }
 
-  /** Checks if a path is a leaf element
+  private def validateErrorColumnName(errorColumnName: String): Unit = {
+    if (errorColumnName.contains('.')) {
+      throw new IllegalArgumentException(s"Error columns should be at the root schema level. " +
+        s"Value '$errorColumnName' is not valid.")
+    }
+  }
+
+  /**
+    * Checks if a path is a leaf element
     * Basically it is just a slightly more efficient version of path.length == 1
     *
     * @param path A path to an element of a struct (e.g. company.employee.firstName)
     * @return true if a path consists only of 1 element meaning it is the leaf element of the input column path
     */
   private def isLeafElement(path: Seq[String]): Boolean = path.lengthCompare(2) < 0
+
+  /**
+    * Converts from a simple transformation to en extended transformation.
+    * In the extended expression a function for getting columns by fully qualified names is ignored.
+    *
+    * @param transformation A transformation function that takes a column and returns a column.
+    * @return An extended transformation that takes a column and a function hat returns columns from fully qualified
+    *         names and returns a column.
+    */
+  private def toExtendedTransformation(transformation: TransformFunction): ExtendedTransformFunction = {
+    (field, _) => transformation(field)
+  }
+
+  /**
+    * The method takes a field and a parent field, both fully qualified. The method does the following:
+    * <ul>
+    *   <li>If the input field is a child of the parent field the path is split into parent and child parts and these
+    *   parts are returned as a pair.</li>
+    *   <li>If the input field is not a child of the parent field the method returns a pair of empty string and
+    *   the full qualified input field name.</li>
+    * </ul>
+    *
+    * @param field       A fully qualified input field name
+    * @param parentField A parent field to split in case the input field is its child
+    * @return A pair consisting of the parent and the child field parts
+    */
+  private[transformations] def splitParentField(field: String, parentField: String): (String, String) = {
```

Comment:
Question: Why `private`?

Reply:
I didn't want to bother with validations so make it for my particular use case. But yeah, it can be promoted to `SchemaUtils` later probably.